tests/src/testresultfunction.cpp

```diff
@@ -0,0 +1,219 @@
+  TestSuite t("Test read in grid");
+  auto epsilon = precision == Dune::Vtk::DataTypes::FLOAT32 ? 1e-4 : 1e-8;
+
+  using Grid = Dune::UGGrid<2>;
+  Dune::GridFactory<Grid> factory;
+
+  Dune::Vtk::VtkReader reader{factory};
+  reader.read(fileName);
+  auto grid     = reader.createGrid();
+  auto gridView = grid->leafGridView();
+
+  t.check(gridView.size(0) == 4) << "GridView should have 4 elements, but has " << gridView.size(0) << " elements";
+
+  // get point Data
+  auto stressData = reader.getPointData("linearStress");
+
+  t.check(stressData.numComponents() == 3) << "Num components should be 3, but is " << stressData.numComponents();
+  t.check(stressData.dataType() == precision) << "Precision is wrong";
+
+  auto localStressFunction = localFunction(stressData);
+  auto results             = expectedStressValues(precision, native);
+  for (int i = 0; const auto& ele : Dune::elements(gridView)) {
+    localStressFunction.bind(ele);
+    auto refEle = Dune::referenceElement(ele);
+    for (const auto c : Dune::range(ele.geometry().corners())) {
+      auto corner       = refEle.geometry<2>(c).center();
+      auto stressInFile = localStressFunction(corner);
+      for (const auto j : Dune::range(3)) {
+        t.check(Dune::FloatCmp::eq(stressInFile[j], results[i], epsilon))
+            << "Result read in is " << stressInFile[j] << "but should be " << results[i];
+        ++i;
+      }
+    }
+  }
+
+  return t;
+}
+
+auto runTestCase() {
+  TestSuite t("Test ResultFunction");
+  std::string fileName = "ResultFunctionTest";
+
+  using Grid     = Dune::UGGrid<2>;
+  using GridView = Grid::LeafGridView;
+
+  constexpr double Lx                                    = 4.0;
+  constexpr double Ly                                    = 4.0;
+  const Dune::FieldVector<double, 2> bbox                = {Lx, Ly};
+  const std::array<unsigned int, 2> elementsPerDirection = {2, 2};
+
+  auto grid     = Dune::StructuredGridFactory<Grid>::createCubeGrid({0, 0}, bbox, elementsPerDirection);
+  auto gridView = grid->leafGridView();
+
+  using namespace Dune::Functions::BasisFactory;
+  auto basis = Ikarus::makeBasis(gridView, power<2>(lagrange<1>()));
+
+  Ikarus::DirichletValues dirichletValues(basis.flat());
+  dirichletValues.fixBoundaryDOFs([&](auto& dirichletFlags, auto&& localIndex, auto&& localView, auto&& intersection) {
+    if (std::abs(intersection.geometry().center()[1]) < 1e-8)
+      dirichletFlags[localView.index(localIndex)] = true;
+  });
+
+  auto vL      = []([[maybe_unused]] auto& globalCoord, auto& lamb) { return Eigen::Vector2d{0, -1}; };
+  auto skills_ = Ikarus::skills(Ikarus::linearElastic({.emodul = 100, .nu = 0.2}), Ikarus::volumeLoad<2>(vL));
+
+  using LinearElastic = decltype(Ikarus::makeFE(basis, skills_));
+  std::vector<LinearElastic> fes;
+
+  for (auto&& element : elements(gridView)) {
+    fes.emplace_back(Ikarus::makeFE(basis, skills_));
+    fes.back().bind(element);
+  }
+
+  /// Create a sparse assembler
```

Comment:
So much boiler plate here... We really have to refactor the tests somehow. I.e. it would be nice to test the FEs with some dummy Grid and dummy Basis. Here so much construction happens.

Reply:
Yes, i can create something for that